src/Scientist/Internals/Experiment.cs

```diff
@@ -10,7 +10,25 @@ internal class Experiment<T, TClean> : IExperiment<T, TClean>, IExperimentAsync<
 
         private static readonly Func<Task<bool>> _alwaysRun = () => Task.FromResult(true);
         private static readonly Action<Operation, Exception> _alwaysThrow
-            = (operation, exception) => { throw exception; };
+            = (operation, exception) =>
+            {
+                string exceptionMessage;
+                switch (operation)
```

Comment:
I see your point, didn't think about that. I've moved the check to a simple `IsDefined`, although, yes, we should only be using defined enums within the code base.